java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -93,22 +93,25 @@ public Pop(ProgramState programState, List<SymbolicValue> result) {
   private SymbolicValue exitSymbolicValue;
   final PMap<Symbol, SymbolicValue> values;
   final PMap<SymbolicValue, PMap<Class<? extends Constraint>, Constraint>> constraints;
+  final PMap<SymbolicValue, Symbol> lastAssociatedSymbol;
 
   private ProgramState(PMap<Symbol, SymbolicValue> values, PMap<SymbolicValue, Integer> references,
-    PMap<SymbolicValue, PMap<Class<? extends Constraint>, Constraint>> constraints, PMap<ProgramPoint, Integer> visitedPoints,
-    PStack<SymbolicValue> stack, SymbolicValue exitSymbolicValue) {
+                       PMap<SymbolicValue, PMap<Class<? extends Constraint>, Constraint>> constraints, PMap<ProgramPoint, Integer> visitedPoints,
+                       PStack<SymbolicValue> stack, SymbolicValue exitSymbolicValue,
+                       PMap<SymbolicValue, Symbol> lastAssociatedSymbol) {
     this.values = values;
     this.references = references;
     this.constraints = constraints;
     this.visitedPoints = visitedPoints;
     this.stack = stack;
     this.exitSymbolicValue = exitSymbolicValue;
+    this.lastAssociatedSymbol = lastAssociatedSymbol;
     constraintSize = 3;
   }
   private ProgramState(Symbol symbol, PMap<Symbol, SymbolicValue> values, PMap<SymbolicValue, Integer> references,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)